co-authors-plus.php

```diff
@@ -600,22 +600,23 @@ public function update_author_term_post_count( $term ) {
 		$query = "SELECT COUNT({$wpdb->posts}.ID) FROM {$wpdb->posts}";
 
 		$query .= " LEFT JOIN {$wpdb->term_relationships} ON ({$wpdb->posts}.ID = {$wpdb->term_relationships}.object_id)";
-		$query .= " LEFT JOIN {$wpdb->term_taxonomy} ON ( {$wpdb->term_relationships}.term_taxonomy_id = {$wpdb->term_taxonomy}.term_taxonomy_id )";
-
-		$having_terms_and_authors = $having_terms = $wpdb->prepare( "{$wpdb->term_taxonomy}.term_id = %d", $term->term_id );
-		if ( 'wpuser' == $coauthor->type ) {
-			$having_terms_and_authors .= $wpdb->prepare( " OR {$wpdb->posts}.post_author = %d", $coauthor->ID );
-		}
-
+		$query .= " LEFT JOIN {$wpdb->terms} ON ({$wpdb->term_relationships}.term_taxonomy_id = {$wpdb->terms}.term_id)";
+		// phpcs:ignore WordPressVIPMinimum.Variables.RestrictedVariables.user_meta__wpdb__users
+		$query .= " LEFT JOIN {$wpdb->users} ON ({$wpdb->posts}.post_author = {$wpdb->users}.ID)";
+	
 		$post_types = apply_filters( 'coauthors_count_published_post_types', array( 'post' ) );
 		$post_types = array_map( 'sanitize_key', $post_types );
 		$post_types = "'" . implode( "','", $post_types ) . "'";
 
-		$query .= " WHERE ({$having_terms_and_authors}) AND {$wpdb->posts}.post_type IN ({$post_types}) AND {$wpdb->posts}.post_status = 'publish'";
-
-		$query .= $wpdb->prepare( " GROUP BY {$wpdb->posts}.ID HAVING MAX( IF ( {$wpdb->term_taxonomy}.taxonomy = '%s', IF ( {$having_terms},2,1 ),0 ) ) <> 1 ", $this->coauthor_taxonomy ); //phpcs:ignore
```

Comment:
Can you tell me your reasoning here behind removing the `HAVING MAX( IF( IF (` section of the query?

Reply:
My understanding of the `HAVING` statement is to further filter results of a previous `GROUP BY` statement. 

For example, I could group the salaries of all employees with a `GROUP BY` statement and then using a `HAVING` statement to only get the salaries that fulfil additional criteria.

In this particular case here, I already group by post ID. Looking at https://codex.wordpress.org/Database_Description#Indexes_6, I see that the post ID is the primary key of the wp_posts table. Therefore, the ID is already unique. Adding a `HAVING` statement after the `GROUP BY` statement is therefore obsolete in my opinion, unless I missed something in this case.